### PR TITLE
Add LXQt 0.12.0 Release

### PR DESCRIPTION
**Pull request by dracwyrm**
I wanted to get this posted for reviews and didn't have time to properly write out the commit messages as there's 29 of them. so I'll fix them later.

### CODE REVIEW

**Comment by blueness** [comment]
I have some interest in keeping lxqt going.  Currently the project is pretty dead.  Rather than interviewing these individually can you tell me which ones needed a significant change on version bump?

**Comment by dracwyrm** [comment]
@blueness I changed quite a few ebuilds from the previous ones. Basically, these need testing on other systems and ebuild inspections. I think I cleaned up everything, but I'm pretty sure there's somethings I missed, like adding some := or whatnot.

They all work and I used it. There's a bug where right-clicking on the panel brings up the menu at the top of the screen instead of where your mouse is. The quick fix is to got to settings and change the location of the panel then back again. I want to check upstream for good patches. There's already one upstream patch and several sed statements that reproduce two patches which are one word changes, so it wasn't worth a patch.

**Comment by dracwyrm** [comment]
I found a fix for the popup menus, so pushed a new update.

**Comment by blueness** [comment]
I did some spot checking and there's nothing subtle that I can see, mostly just dependency stack.  Why do you think you might have missed some slot/subslot operators?  Just look at whatever libraries your depending on, and see if := is appropriate.

**Comment by a17r** [comment]
I see lots of !! hard blockers there, what's the rationale? File collisions are not.

**Comment by dracwyrm** [comment]
I knew I missed something. I was half asleep when I wrote the ebuilds. I'll try with single ! to see if that works. There will be file collisions if it doesn't.

**Comment by blueness** [comment]
Please don't submit sloppy work and expect others to do the cleanup.  I'd like to see this get in, but I'm not prepared to commit it without the confidence it was done right.

**Comment by dracwyrm** [comment]
@a17r @blueness I went through and added in appropriate metadata tot he commits this morning since it's my day off and finally have time to work on this. The dependencies should be right now.

I have tested this on AMD64 and ARM64 platforms, and it works.

The main thing that needs testing is the new package pavucontrol, which is the new default.

The menu-cache update fixes CVE-2017-8933: https://bugs.gentoo.org/618620. This needs to be tested with all things that depend on it so it can be a candidate for stabilization in a months time. The security team will need to evaluate this package to make sure the flaw is fixed.

**Comment by Chiitoo** [comment]
Regarding your question via IRC today (just arrived when you left!), yeah, it seems I used a double (!!) mask at least here:

https://gitweb.gentoo.org/proj/qt.git/tree/lxqt-base/lxqt-themes/lxqt-themes-9999.ebuild#n23

Unfortunately, I don't really remember the details of my testing on that one, but as others have pointed out that it should not be necessary, I probably missed something?

I don't have any systems with PulseAudio, and I'm currently on Qt 5.11.9999, which adds its own fun to the mix, so I can't test 'pavucontrol' too well any time soon...

Going through the ebuilds here, I did not come up with anything to comment on really, aside from maybe 'https' being an option for the home page too.  :]

I seem to remember upstream not doing point releases for every package at some point (pcmanfm-qt-0.11.3 for example), which is why I opted for depending via '>=' instead of '~'.  I don't know if that was smart, or mad, both, or other...

Thanks!

**Comment by charIes17** [comment]
For lxde-base/menu-cache, see https://github.com/gentoo/gentoo/pull/5355

**Comment by dracwyrm** [comment]
@charIes17 Sorry. Didn't notice your PR for menu-config. The only thing I noticed with your ebuild is that for 1.1.0 they switched to LGPL-2.1+ license as per https://github.com/lxde/menu-cache/blob/master/COPYING.

**Comment by blueness** [comment]
When the changes settles down, I'll test build on amd64 and x86 and push.  Let me know when you think its ready.

**Comment by dracwyrm** [comment]
I thought about how to take into account what Chiitoo said about the the point releases. The solution I updated the ebuilds with should work and be easy to update for future releases. :) I hope this fits with Gentoo policy and QA.

@blueness The ebuilds should be ready for testing now. Thanks.

File: lxde-base/menu-cache/menu-cache-1.1.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	x11-libs/libfm-extra:=
+"
+DEPEND="${RDEPEND}
+	sys-devel/gettext
+	virtual/pkgconfig
+"
+
+src_install() {
+	default
+	find "${ED}" \( -name "*.a" -o -name "*.la" \) -delete
```
Comment (SoapGentoo) [inline lxde-base/menu-cache/menu-cache-1.1.0.ebuild]: try and see if you can specify `--disable-static` to `econf` so you dont need to unnecesaarily build it

File: lxqt-base/lxqt-config/lxqt-config-0.12.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+DEPEND="${RDEPEND}
+	>=dev-util/lxqt-build-tools-0.4.0
+	dev-qt/linguist-tools:5
+"
+
+mycmakeargs=( -DPULL_TRANSLATIONS=OFF )
+
+src_install(){
+	cmake-utils_src_install
+	doman man/*.1 liblxqt-config-cursor/man/*.1 lxqt-config-appearance/man/*.1
```
Comment (SoapGentoo) [inline lxqt-base/lxqt-config/lxqt-config-0.12.0.ebuild]: try and seeif you can fix this in the cmake so it's upstreamed?

File: lxde-base/menu-cache/menu-cache-1.1.0.ebuild (hunk 1 above)
Comment (dracwyrm) [reply lxde-base/menu-cache/menu-cache-1.1.0.ebuild]: I tried with --disable-static and the difference is that only the .a file is not installed, the .la file still is, so it doesn't disable static.

File: lxqt-base/lxqt-config/lxqt-config-0.12.0.ebuild (hunk 2 above)
Comment (dracwyrm) [reply lxqt-base/lxqt-config/lxqt-config-0.12.0.ebuild]: I'll see what I can find.

**Comment by DraGiuS** [comment]
And so ?

**Comment by blueness** [comment]
Let me get a signoff by the qt project becomes some of ebuilds touch packages that belong to qt project.

**Comment by Arfrever** [comment]
Deleting `x11-libs/qtermwidget/qtermwidget-9999.ebuild` and `x11-terms/qterminal/qterminal-9999.ebuild` does not seem to make sense.

**Comment by a17r** [comment]
ping

**Comment by blueness** [comment]
@Arfrever objections were never addressed and now there is a conflict.  also i can't commit over packages that belong to the qt project and there are a few of those.  so i'm giving up on this unless someone really wants to put the effort in and create a clean PR.

**Comment by Dessa** [comment]
@blueness im going to split the PR up *a bit*, leaving the qterm* stuff out if i can (since it needs newer lxqt-build-tools but this newer version *might* break 0.11.0 i will need to test that first.
for libqtxdg i have created #7957 since it doesn't break things for me when present outside of lxqt 0.12

**Comment by blueness** [comment]
@Dessa yeah that will help

File: lxqt-base/lxqt-config/lxqt-config-0.12.0.ebuild (hunk 2 above)
Comment (Dessa) [reply lxqt-base/lxqt-config/lxqt-config-0.12.0.ebuild]: i have notified upstream about this. im not sure if that will hit 0.13 though

**Comment by Dessa** [comment]
at this point, i don't think a non-monolithic bump will be feasible.

a few packages need qt-5.11 fixes, which will be available in lxqt-0.13
i have no ETA on the release, but it should be soon as they plan to be in line with qt-5.11 release

**Comment by jubalh** [comment]
Yes 0.13.0 is out since a while, and builds fine with Qt 5.11/5.11.1

**Comment by khumarahn** [comment]
What is the current state? Does anybody have a set of ebuilds which work for the new lxqt?

**Comment by Arfrever** [comment]
The last member of LXQt project has resigned on 2018-07-06:
https://archives.gentoo.org/gentoo-dev/message/889aefc22008801fae21fa5c8024a086
So new maintainers will need to be found.

**Comment by dracwyrm** [comment]
I'm trying to submit a PR for 0.13.0 but I'm having issues with GitHub and credentials.

**Pull request closed without merging**